python/lsst/ts/watcher/rules/mtdome_capacitor_banks.py

```diff
@@ -0,0 +1,119 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MTDomeCapacitorBanks"]
+
+import typing
+
+from lsst.ts import salobj
+from lsst.ts.xml.enums.Watcher import AlarmSeverity
+
+from ..base_rule import AlarmSeverityReasonType, BaseRule, NoneNoReason
+from ..remote_info import RemoteInfo
+
+REASON_DICT: dict[str, str] = {
+    "doorOpen": "Open door",
+    "fuseIntervention": "Broken fuse",
+    "highTemperature": "High temperature",
+    "lowResidualVoltage": "Low residual voltage",
+    "smokeDetected": "Smoke",
+}
+
+
+class MTDomeCapacitorBanks(BaseRule):
+    """Monitor the MTDome capacitor banks for any alarming states.
+
+    Parameters
+    ----------
+    config : `types.SimpleNamespace`
+        Rule configuration, as validated by the schema.
+    log : `logging.Logger`, optional
+        Parent logger.
+    """
+
+    def __init__(self, config, log=None):
+        remote_name = "MTDome"
+        remote_index = 0
+        remote_info_list = [
+            RemoteInfo(
+                name=remote_name,
+                index=remote_index,
+                callback_names=["evt_capacitorBanks"],
+                poll_names=[],
+            )
+        ]
+        super().__init__(
+            config=config,
+            name=f"MTDomeCapacitorBanks.{remote_name}:{remote_index}",
+            remote_info_list=remote_info_list,
+            log=log,
+        )
+
+    @classmethod
+    def get_schema(cls):
+        # No schema necessary for this rule.
+        return None
+
+    def compute_alarm_severity(
+        self, data: salobj.BaseMsgType, **kwargs: typing.Any
+    ) -> AlarmSeverityReasonType:
+        """Compute and set alarm severity and reason.
+
+        Parameters
+        ----------
+        data : `salobj.BaseMsgType`
+              Message from the topic described by topic_callback.
+        **kwargs : `dict` [`str`, `typing.Any`]
+            Keyword arguments. If triggered by `TopicCallback` calling
+            `update_alarm_severity`, the arguments will be as follows:
+
+            * topic_callback : `TopicCallback`
+              Topic callback wrapper.
+
+        Returns
+        -------
+        None, if no change or unknown, or a tuple of two values:
+
+        severity: `lsst.ts.idl.enums.Watcher.AlarmSeverity`
+            The new alarm severity.
+        reason : `str`
+            Detailed reason for the severity, e.g. a string describing
+            what value is out of range, and what the range is.
+            If ``severity`` is ``NONE`` then this value is ignored (but still
+            required) and the old reason is retained until the alarm is reset
+            to ``nominal`` state.
+
+        Notes
+        -----
+        You may return `NoneNoReason` if the alarm state is ``NONE``.
+        """
+        severity, reason = NoneNoReason
+
+        reason_list: list[str] = []
+        for attribute, value in vars(data).items():
```

Comment:
I think it would be simpler to just loop over the `REASON_DICT ` dictionary and use `getattr` no? Also, the way it is done here, there's a chance that the xml is updated to rename a field and we will never get a failure to indicate the change.